### DISCUSSION TITLE
mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/

### DISCUSSION THREAD

**Discussion by giscus** [squidfunk/mkdocs-material#7462] [_]
# mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/

We are transforming Material for MkDocs to ensure its community continues to thrive - building the best tools for creating professional documentation.

https://squidfunk.github.io/mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/

**Comment by pawamoy** [comment]
Congratulations to the three of you! It's very cool to see how much care and thinking you put into the products you build :slightly_smiling_face:

**Reply by Guts** [reply]
I agree, thanks for the insights and looking forward for your exciting work!

**Comment by alonsomoya** [comment]
Really excited about what's coming ahead! Thanks a ton for your efforts to give enterprise users a voice. I'm convinced that `mkdocs-material` has the potential to transform the documentation culture of many organizations!

**Comment by leskaiser** [comment]
Congratulation and thanks for work

**Comment by ojacques** [comment]
Thank you for sharing a detailed update to where you are going. I am happy to see the commitment to Open Source, while still keeping the possibility to financially support, for those who can. 
Can’t wait to continue using and advocating doc-as-code everywhere I go!

**Comment by rachel-rigdon-sp** [comment]
Thanks for the thoughtful update! I especially appreciate the emphasis on search and increasing scalability. I'm really excited to see how everything develops!

**Comment by AtticusZeller** [comment]
🎉🎉🎉

**Comment by Andre601** [comment]
I honestly was - and quite honestly still am - on mixed feelings about the chat bot/LLM integration.
It may sound and look like a good idea, but I also believe that the current oversaturation of LLMs everywhere has started to "poison the well" so to speak (with ironocally enough LLMs poisoning each other in the process by learning from each others (often times faulty) data).

So I do hope that you guys will take your time with this until the entire AI-mania dies down eventually, so that this project doesn't become yet another one with AI implementation.

**Comment by adwaitlearn** [comment]
Would love to see the Editing and collaboration being available in mkdocs as it would certainly help a lot of teams.

I know its a huge task...Would it be possible to give an estimated ETA in months or probably year/s?

**Reply by squidfunk** [reply]
I'm sorry, we can't currently give an ETA. However, we're currently working on the foundation to make it happen 🚀

**Comment by mablospate** [comment]
simply amazing

**Comment by zakibashir** [comment]
Thank you

**Comment by ryneches** [comment]
I'm looking forward to the new and shiny mkdocs, but I'm a little confused about how to contribute code in the mean time.

I just want to post Jupyter notebooks as blog posts. This is an extremely common use case in data science, and at the moment, there are basically zero platforms that support this that are being actively maintained. That's... kinda terrible, I think! With just a little bit of code to hand off the metadata from the [mkdocs-jupyter](https://github.com/danielfrg/mkdocs-jupyter) plugin by @danielfrg to mkdocs-material's built in blog plugin, I have exactly what I want. However, mkdocs is the fifth platform that I've had to hack Jupyter support into, only to have the effort go to waste as the maintainers move on to other things. I'm frankly *exhausted*.

The news of a ground-up rewrite does not fill me with enthusiasm to contribute. Please tell me I am wrong? Please?

**Reply by mablospate** [reply]
I think, although I don't share your use case, that your concerns and complaints are completely justified

**Reply by squidfunk** [reply]
Please understand, that since we're currently in feature freeze, because of the foundational work we're at, we cannot accept PRs that add new features or interations. However, what we can already say is that integration with third-party frameworks like Jupyter will be much simpler than it currently is. As we've described in the post, we've reached the end of trying to make plugin interop possible. I've written 12 plugins myself (all of which are bundled with Material), and believe me, it's hard to make them stable and work together. Thus, we share your frustration and took this as an opportunity. The current architecture of MkDocs just doesn't allow for creating stable contracts between plugins. We believe we now have a design that will allow for plugins to provide explicit contracts, which will unlock a whole lot of new possibilities, but we still need some more time, since this is a very complex undertaking, and we are a very small team. 

However, be assured that compatibility and/or easy migration are one of our goals. Thank you for your patience.

**Reply by ryneches** [reply]
That's great to hear!

Please understand that I'm not trying to aim any negativity at you folks. I just don't want to put my attention and energy into something that's been abandoned by the core developers. Web technologies have wasted so much of everyone's time, and I'm tired. I'm a researcher, not a web developer. I need something stable -- in the social sense. Code can be fixed when it breaks, but once a community gives up on a project, there's no way a user can keep it alive by themselves. 

I don't want to get involved in high level design, but if you have a spec and some milestones, I (and probably a fair few other folks) would be delighted to start bolting things together to see what happens.

If you're committed to the project, then I want to help. Please consider this to be a friendly nudge from someone who wants to see you succeed.

**Comment by egat98** [comment]
Creator
